modules/stochastic_tools/include/state_sim/model/diagrams/StateSimDiagramEval.h

```diff
@@ -0,0 +1,55 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMDIAGRAMEVAL_H
+#define STATESIMDIAGRAMEVAL_H
+
+#include "StateSimDiagram.h"
+#include "StateSimBitset.h"
+#include "StateSimState.h"
+#include <unordered_map>
+#include <string>
+#include <vector>
+
+/**
+ * Eval diagrams are model diagrams that can be evaluated, returning a boolean value according to which
+ * of their states are in the current state list. (Component and System Diagrams)
+ */
+class StateSimDiagramEval : public StateSimDiagram
+{
+protected:
+  DIAGRAM_TYPE_ENUM _diag_type;
+  std::unordered_map<int, StateSimState *> _ok_states;
+  //std::unordered_map<int, StateSimState *> _single_state_group;
+  std::unordered_map<int, StateSimState *> _fail_states;
+
+private:
+
+
+public:
```

Comment:
Usually in C++ public members come first and protected/private come last. This pattern while not forbidden is unusual. I'd prefer to see this ordering swapped throughout your code.